parsons/ngpvan/bulk_import.py

```diff
@@ -24,3 +28,123 @@ def get_bulk_import_resources(self):
         r = self.connection.get_request(f'bulkImportJobs/resources')
         logger.info(f'Found {len(r)} bulk import resources.')
         return r
+
+    def get_bulk_import_job(self, job_id):
+        """
+        Get a bulk import job status.
+
+        `Args:`
+            job_id : int
+                The bulk import job id.
+        `Returns:`
+            dict
+                The bulk import job
+        """
+
+        r = self.connection.get_request(f'bulkImportJobs/{job_id}')
+        logger.info(f'Found bulk import job {job_id}.')
+        return r
+
+    def get_bulk_import_mapping_types(self):
+        """
+        Get bulk import mapping types.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.connection.get_request('bulkImportMappingTypes'))
+        logger.info(f'Found {tbl.num_rows} bulk import mapping types.')
+        return tbl
+
+    def get_bulk_import_mapping_type(self, type_name):
+        """
+        Get a single bulk import mapping type.
+
+        `Args:`
+            type_name: str
+        `Returns`:
+            dict
+                A mapping type json
+        """
+
+        r = self.connection.get_request(f'bulkImportMappingTypes/{type_name}')
+        logger.info(f'Found {type_name} bulk import mapping type.')
+        return r
+
+    def post_bulk_import(self, tbl, url_type, resource_type, mapping_types,
+                         description, **url_kwargs):
+        # Internal method to post bulk imports.
+
+        # Move to cloud storage
+        file_name = str(uuid.uuid1())
+        url = cloud_storage.post_file(tbl, url_type, file_path=file_name + '.zip', **url_kwargs)
+        logger.info(f'Table uploaded to {url_type}.')
+
+        # Generate request json
+        json = {"description": description,
+                "file": {
+                    "columnDelimiter": 'csv',
+                    "columns": [{'name': c} for c in tbl.columns],
+                    "fileName": file_name + '.csv',
+                    "hasHeader": "True",
+                    "hasQuotes": "False",
```

Comment:
It looks like the Python `csv.writer` (via `petl`) is putting quotes in when necessary.. which appears to be the default (`csv.QUOTE_MINIMAL`).

```python
from parsons import Table
tbl = Table([{'one': 'one', 'two': 'two,'}])
tbl.to_csv()
```

Produces:

```
one,two
one,"two,"
```

It might be best to write out the CSV file with `csv.QUOTE_ALL` and then set `hasQuotes` to `True` always.

`quoting` is a keyword arg for the `csv.writer`, which you can pass into the `petl.tocsv`.. I'm not sure though how to wire that up through the `post_file` call above.

Reply:
Fixed.